src/main/java/com/google/sps/servlets/SearchServlet.java

```
@@ -26,25 +43,83 @@
-      recipes.add(recipe);
+    int error = 0;
+    List<Recipe> recipesList = new ArrayList<>();
+    String titleToMatch = request.getParameter("searchterm");
+    Index index = getIndex();
+    Cursor responseCursor = Cursor.newBuilder().build();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    QueryOptions options = QueryOptions
+      .newBuilder()
+      .setLimit(10)
+      .setFieldsToSnippet("title")
+      .setCursor(responseCursor)
+      .setSortOptions(
+        SortOptions
+          .newBuilder()
+          .addSortExpression(
+            SortExpression
+              .newBuilder()
+              .setExpression("title")
+              .setDirection(SortExpression.SortDirection.ASCENDING)
+              .setDefaultValue("")
+          )
+      )
+      .build();
+
+    Query query = Query
+      .newBuilder()
+      .setOptions(options)
+      .build(titleToMatch.toLowerCase());
+
+    Results<ScoredDocument> results = index.search(query);
+
+    for (ScoredDocument entity : results) {
+      try {
+        String id = entity.getId();
+        Entity recipeEntity = datastore.get(
+          KeyFactory.createKey("Recipe", Long.parseLong(id))
+        );
+
+        String name = (String) recipeEntity.getProperty("title");
+
+        Recipe recipe = new Recipe();
+        String description =
+          "Lorem quam dolor dapibus ante, sit amet pellentesque turpis lacus eu ipsum. Duis quis mi ut tortor interdum efficitur quis at mi. Pellentesque quis mauris vel ligula commodo scelerisque. In vulputate quam nisl, vel sagittis ipsum molestie quis. Suspendisse quis ipsum a sem aliquam euismod mattis sed metus.";
+
+        recipe.setId(Long.parseLong(id));
+        recipe.setName(name);
+        recipe.setImage((String) recipeEntity.getProperty("imgURL"));
+        recipe.setDescription(description);
+
+        recipesList.add(recipe);
+      } catch (EntityNotFoundException e) {
+        error = 1;
```

You should throw an exception if this is not supposed to happen.